### PR TITLE
feature/minimum ts

### PR DESCRIPTION
**Pull request by alanrsoares**
quality of life improvements:

* eslint checks
* prettier auto formatting + import sorting with `prettier-plugin-sort-imports` and tailwind class sorting with `prettier-plugin-tailwindcss`
* code formatting on commits with pretty-quick + husky
* commitlint with conventional commits + husky
* rome lint autofixes

next steps:

* review state management:
  * most reducers are only used to store a single value, usually returns from apis
* more robust api clients:
  * leverage code generation for rest api clients based on openApi/swagger
  * use `rtk/query` or `react-query` for better data-fetching/caching patterns
* replace `lodash` with `rambda` for stricter functional style, better typechecks
* replace `moment` with `date-fns` for better tree-shaking, producing smaller bundles

image

### CODE REVIEW

File: src/pages/_app.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          content="/icons/mstile-150x150.png"
+        />
+        <meta name="theme-color" content="#050707" />
+      </Head>
+      <Provider store={store}>
+        <Layout>
+          <div id="portal" />
+          <Component {...pageProps} />
+        </Layout>
+        <div className="grid-ceols-3 grid-cols-2 grid-cols-4" />
```
Comment (pjenviri) [inline src/pages/_app.js]: a typo; the className should be `grid-cols-3`.

Comment (alanrsoares) [reply src/pages/_app.js]: good catch, this was a rogue keybinding, haha

Comment (alanrsoares) [reply src/pages/_app.js]: however, this div doesn't make sense, which class is it supposed to use? there's no breakpoint distinction

Comment (alanrsoares) [reply src/pages/_app.js]: just removed it

**Review by pjenviri — APPROVED** [review]

**Pull request merged**